### ISSUE TITLE
Re-consider mocked test

### ISSUE THREAD

**Issue by gunnarmorling** [mincong-h/gsoc-hsearch#136]
Instead of doing all the mock, let's re-use the real code and mock only some parts as needed. Ideally, we can work with a single mocking lib (Mockito) instead of two.

**Comment by mincong-h** [comment]
Tests need to be refactored, especially [`PartitionMapperTest.java`][1]. In this class, 15 objects have been mocked and it leads to very poor code for user-understanding. There're several solutions for the refactoring with pros and cons:

### Solution 1 - Mocking (current test)

* ✅ No additional dependencies on other classes
* ✅ Do not need any external data and environment settings in general
* ❌ Really complicated to understand, test logics lose among the tones of assumptions
* ❌ Failed on mocking conversion between String and primitive types

### Solution 2 - Test the partition plan

Instantiate a `PartitionMapper` inside the test, inject every attributes and run its method `mapPartitions()`. Once finished, we receive the returned object of type `PartitionPlan` and test the content inside this instance. 

* ❌ Since to be the best choice, but I don't know how to do the injection to this object myself.

### Solution 3 - Test a dummy chunk

Test a dummy chunk with dummy JSL file (job XML) and dummy chunk components (ItemReader, ItemProcessor, ItemWriter). And then we test whether the job runs correctly and test its metrics at the end of the job.

* ✅ JSL is more simple to understand
* ✅ Chunk components are more simple to understand
* ❌ Difficult the make sure that the dummy components are aligned with the real ones.
* ❌ Cannot receive the partition plan. So only the final metrics can be tested. We cannot test the exact value of different objects in partitions explicitly.

### Solution 4 - Test a real chunk

* ✅ Not inventing the wheel
* ❌ Cannot receive the partition plan. So only the final metrics can be tested. We cannot test the exact value of different objects in partitions explicitly.

[1]: https://github.com/mincong-h/gsoc-hsearch/blob/217d560424171e83864dfe55f6d08d955baeb763/core/src/test/java/org/hibernate/search/jsr352/PartitionMapperTest.java

**Comment by mincong-h** [comment]
Finally I've used the _Solution 2_. After having asked [a question on Stack Overflow][1] yesterday, I understand better how to do it and then I refactored the tests. Now these tests use a real `EntityManagerFactory` instead of mocked one, so there're much less assumptions / mocks. Batch properties are passed to the instance using a protected args-constructor.

Previously, PowerMock need to be used for mocking `ContextHelper.class`. After the refactoring, this isn't needed anymore. 💯 

[1]: http://stackoverflow.com/questions/39618765/injecting-private-fields-having-inject-during-test
